libs/ngxtension/inject-destroy/src/inject-destroy.ts

```diff
@@ -0,0 +1,35 @@
+import { DestroyRef, assertInInjectionContext, inject } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+
+/**
+ * Injects the `DestroyRef` service and returns a `ReplaySubject` that emits
+ * when the component is destroyed.
+ *
+ * @throws {Error} If no `DestroyRef` is found.
+ * @returns {ReplaySubject<void>} A `ReplaySubject` that emits when the component is destroyed.
+ *
+ * @example
+ * // In your component:
+ * export class MyComponent {
+ *   private destroy$ = injectDestroy();
+ *
+ *   getData() {
+ *     return this.service.getData()
+ *       .pipe(takeUntil(this.destroy$))
+ *       .subscribe(data => { ... });
+ *   }
+ * }
+ */
+export const injectDestroy = () => {
```

Comment:
suggestion: let's have `injectDestroy` accept an `Injector` so it is consistent with the rest.